ext/NearestNeighborsExt.jl

```diff
@@ -0,0 +1,130 @@
+    return id, sep
+end
+function knn(tree::KDTree, coords::AbstractArray{<:AbstractSkyCoords}, k::Int, sortres::Bool = false)
+    if isempty(coords)
+        throw(ArgumentError("`coords` provided to `knn` cannot be empty."))
+    end
+    id, sep = knn(tree, map(_convert, vec(coords)), k, sortres)
+    for i in eachindex(sep)
+        @. sep[i] = 2 * asin(sep[i] / 2) # Convert from cartesian separation to radians
+    end
+    return reshape(id, size(coords)), reshape(sep, size(coords))
+end
+
+"""
+    inrange(tree::KDTree, coord::AbstractSkyCoords, seplim::Number)
+Searches for coordinates in the `tree` with angular separations from `coord` less than `seplim`, which must be given in radians. If `tree = KDTree(data)`, returns indices into `data`.
+
+    inrange(tree::KDTree, coords::AbstractArray{<:AbstractSkyCoords}, seplim::Number)
+For each coordinate in `coords`, finds coordinates in `tree` that lie within `seplim` radians.
+"""
+function inrange(tree::KDTree, coord::AbstractSkyCoords, 
+                 seplim::Number)
+    cart_sep = 2 * sin(seplim / 2) # Cartesian distance = seplim
+    return inrange(tree, _convert(coord), cart_sep)
+end
+function inrange(tree::KDTree, coords::AbstractArray{<:AbstractSkyCoords}, 
+                 seplim::Number)
+    cart_sep = 2 * sin(seplim / 2) # Cartesian distance = seplim
+    return inrange(tree, map(_convert, vec(coords)), cart_sep)
+end
+
+"""
+    match(refcoords::AbstractArray{<:AbstractSkyCoords}, 
+          matchcoords::AbstractArray{<:AbstractSkyCoords};
+          nthneighbor::Int = 1)
+*Requires Julia ≥ 1.9 and NearestNeighbors.jl to be loaded (e.g., `using NearestNeighbors`).*
+
+Finds the nearest entries in `refcoords` to the coordinates contained in `matchcoords`. The keyword argument `nthneighbor` determines *which* nearest neighbor to search for; typically this should be `1` when matching one set of coordinates to another. Another common use case is setting `nthneighbor = 2` when matching a catalog against itself to find the nearest neighbor of each coordinate in the same catalog. 
+
+Returns `(id, sep)`, where
+ - `id` is an array containing indices of the coordinates in `refcoords` that matched with the elements of `matchcoords`, and 
+ - `sep` is an array giving the angular separation between the elements of `matchcoords` and the above matches.
+
+Note that this method creates a [`KDTree`](@ref) from `refcoords` and then calls the method below. If you plan to use the same `refcoords` to match to many different `matchcoords`, then you should directly construct `KDTree(refcoords)` and call the method below.
+
+    match(tree::KDTree, matchcoords::AbstractArray{<:AbstractSkyCoords};
+          nthneighbor::Int = 1)
+As above, but uses a pre-constructed `tree::KDTree` rather than creating one from a reference catalog of coordinates.
+"""
+function Base.match(tree::KDTree, matchcoords::AbstractArray{<:AbstractSkyCoords};
```

Comment:
How about defining a new `SkyCoords.match` function instead of punning `Base.match`?

Reply:
> Thanks for the explanation! Let me know if my latest commit addresses this issue appropriately.

It does, thanks.

> Do you think we should export our `match` function or not?

I wouldn't, otherwise you wouldn't be able to call `Base`'s `match` unqualified in the same scope. If users _really_ want to (e.g. because they're going match loads of coordinates) they can always import the symbol with something like
```julia
using SkyCoords: match
```

This is my personal opinion, but I think `SkyCoords.match` is nicer than `match_coords`, namespacing is done via the module owning the function, rather than adding extra information to the function name. And this function being different from `Base.match` avoids giving troubles to the compiler.